### DISCUSSION TITLE
Mind map

### DISCUSSION THREAD

**Discussion by pleshevskiy** [terrastruct/d2#261] [Feature Requests]
I want to use d2 as a mindmap to make it easy to structure documentation.

![image](https://user-images.githubusercontent.com/7839004/204640568-d6819ac5-9b74-4c4f-9606-361ac4b9ec4e.png)

**Comment by mandarvaze** [comment]
It would be nice if there was a way to display nodes on both sides of the central theme like [PlantUML's arithmatic notation](https://plantuml.com/imgw/img-f1ee1d1208a1f7a1d8b0522ecbb7c3c7.png) supports.
See [documentation](https://plantuml.com/mindmap-diagram) for details.

**Reply by pleshevskiy** [reply]
Yes, that would be awesome! But for a start, make sure the levels don't stretch to their maximum width. Details here: #254

**Comment by mandarvaze** [comment]
@pleshevskiy Did you try TALA layout as suggested by @alixander ?

**Reply by pleshevskiy** [reply]
No, yet

**Comment by bo-ku-ra** [comment]
It seems to have been achieved in "d3.js".
https://codepen.io/simsmbb/pen/jmXVaz
https://d3js.org/

i don't understand, but will it be realized if d2 supports "dagre-d3"?
https://github.com/dagrejs/dagre-d3

**Reply by alixander** [reply]
this is not the work of dagre.

D3 implements other algorithms too. This one is https://d3-wiki.readthedocs.io/zh_CN/master/Tree-Layout/

https://reingold.co/tidier-drawings.pdf

Should be straightforward to translate that algorithm to Go for D2 as well, and apply it on `shape: tree` or something.

**Reply by alixander** [reply]
their whole source code is only 200-300 lines: https://github.com/d3/d3-hierarchy/blob/main/src/tree.js (didn't look at how much the Node import is doing)
